internal/provider/harness_k3s_resource.go

```diff
@@ -326,6 +341,23 @@ You can access the cluster with something like: "KUBECONFIG=%s kubectl get po -A
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
+func validateKubeletConfig(content string) error {
```

Comment:
I like this validation approach but can we move it closer to its usage (in the `k3s`) package?

probably just copy/pasta the function into a k3s `WithKubeletConfig` `Option`

Reply:
Sure!

Reply:
Updated!